navigation/Routes.js

```diff
@@ -0,0 +1,38 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+
+import { auth } from '../components/Firebase/firebase';
+import navigationTheme from './navigationTheme';
+import AuthStack from './AuthStack';
+import AppStack from './AppStack';
+import { AuthUserContext } from './AuthUserProvider';
+import Spinner from '../components/Spinner';
+
+export default function Routes() {
+  const { user, setUser } = useContext(AuthUserContext);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const authListener = auth.onAuthStateChanged(async authUser => {
+      try {
+        await (authUser ? setUser(authUser) : setUser(null));
+        setIsLoading(false);
+      } catch (error) {
+        console.log(error);
+      }
+    });
+
+    // unsubscribe listener on unmount
+    return () => authListener;
```

Comment:
What is `authListener` doing here, if it's a method to unregister the listener you have to invoke it, right?

Reply:
Yes, here the listener is getting unsubscribed. Is there any other way to unsubscribe it? I am using the same strategy with React Native vanilla apps using `react-native-firebase` package.

Reply:
It's a hard-to-spot issue, but I ran into this one a couple of times when creating hooks for Expo. I think what you actually wanted to do here is `return authListener;`. Once the hook unmounts, the unsubscriber is called ([see this SO](https://stackoverflow.com/questions/37370224/firebase-stop-listening-onauthstatechanged)).

You can also write it as `() => authListener()` ofc., but that's eventually the same thing. You only add an extra method 😄 Maybe it's also good to rename the variable to something like `unsubscribeAuth` or `removeAuthListener`. Should make it a bit more understandable as well.

Reply:
Thanks for sharing this and taking a look at this, @byCedric. I will update it likewise both based on the correct form of returning an unsubscribe function returning from the `onAuthStateChanged`  as well use better naming convention.